File: aurora-springboot/src/main/java/com/aurora/model/vo/ArticleVO.java
package com.aurora.model.vo;


import io.swagger.v3.oas.annotations.media.Schema;
import lombok.*;

import jakarta.validation.constraints.NotBlank;
import java.util.List;

@Data
@Builder
@AllArgsConstructor
@NoArgsConstructor
@Schema(description = "文章")
public class ArticleVO {

    @Schema(name = "id", description = "文章id")
    private Integer id;

    @NotBlank(message = "文章标题不能为空")
    @Schema(name = "articleTitle", description = "文章标题", required = true)
    private String articleTitle;

    @NotBlank(message = "文章内容不能为空")
    @Schema(name = "articleContent", description = "文章内容", required = true)
    private String articleContent;

    @Schema(name = "articleAbstract", description = "文章摘要")
    private String articleAbstract;

    @Schema(name = "articleCover", description = "文章缩略图")
    private String articleCover;

    @Schema(name = "category", description = "文章分类")
    private String categoryName;

    @Schema(name = "tagNameList", description = "文章标签")
    private List<String> tagNames;

    @Schema(name = "isTop", description = "是否置顶")
    private Integer isTop;

    @Schema(name = "isFeatured", description = "是否推荐")
    private Integer isFeatured;

    @Schema(name = "status", description = "文章状态")
    private Integer status;

    @Schema(name = "type", description = "文章类型")
    private Integer type;

    @Schema(name = "originalUrl", description = "原文链接")
    private String originalUrl;

    @Schema(name = "password", description = "文章访问密码")
    private String password;
}

File: aurora-go/internal/vo/article_vo.go
package vo

// ArticleVO 文章发布/编辑请求（对齐Java版ArticleVO）
type ArticleVO struct {
	ID              uint     `json:"id,omitempty"`                   // 编辑时传
	ArticleTitle    string   `json:"articleTitle" binding:"required,max=200"`
	ArticleContent  string   `json:"articleContent" binding:"required"`
	ArticleAbstract string   `json:"articleAbstract,omitempty"`      // 文章摘要
	ArticleCover    string   `json:"articleCover,omitempty"`         // 文章封面
	CategoryName    string   `json:"categoryName,omitempty"`         // 分类名称（对齐Java）
	TagNames        []string `json:"tagNames,omitempty"`             // 标签名称列表（对齐Java）
	Type            int8     `json:"type" binding:"omitempty,oneof=1 2 3"`
	OriginalURL     string   `json:"originalUrl,omitempty" binding:"omitempty,max=500"`
	IsTop           int8     `json:"isTop" binding:"omitempty,oneof=0 1"`
	IsFeatured      int8     `json:"isFeatured" binding:"omitempty,oneof=0 1"`
	Status          *int8    `json:"status" binding:"omitempty,oneof=0 1 2 3"`
	Password        string   `json:"password,omitempty" binding:"omitempty,max=64"` // 密码保护文章
}

// ArticleTopFeaturedVO 置顶/推荐操作（对标Java ArticleTopFeaturedVO）
// 注意: 使用指针类型 *int8，对标Java Integer（对象类型），避免 required 验证将0视为空值
type ArticleTopFeaturedVO struct {
	ID         uint   `json:"id" binding:"required"`
	IsTop      *int8  `json:"isTop" binding:"required,oneof=0 1"`       // 指针类型，对标Java Integer
	IsFeatured *int8  `json:"isFeatured" binding:"required,oneof=0 1"` // 指针类型，对标Java Integer
}
